modules/cdp-engine/src/lib.rs

```diff
@@ -1042,3 +1044,31 @@ impl<T: Config> RiskManager<T::AccountId, CurrencyId, Balance, Balance> for Pall
 fn pick_u32<R: RngCore>(rng: &mut R, max: u32) -> u32 {
 	rng.next_u32() % max
 }
+
+// Allows for homa-lite module to access cdp functionality
+impl<T: Config> UpdateLoan<T::AccountId, Amount> for Pallet<T> {
+	fn get_position(who: &T::AccountId, staking_id: CurrencyId) -> Result<Position, DispatchError> {
+		match loans::Positions::<T>::try_get(staking_id, who) {
+			Ok(val) => Ok(val),
+			Err(_e) => Err(Error::<T>::NoOpenPosition.into()),
+		}
+	}
+
+	fn transfer_collateral_from_loan(currency_id: CurrencyId, to: &T::AccountId, balance: Balance) -> DispatchResult {
+		let module_id = loans::Pallet::<T>::account_id();
+		T::Currency::transfer(currency_id, &module_id, to, balance)?;
+		Ok(())
+	}
+
+	fn swap_position_to_liquid(
+		who: &T::AccountId,
+		staking_id: CurrencyId,
+		liquid_id: CurrencyId,
+		liquid_adjustment: Amount,
+		staking_adjustment: Amount,
+		debit_adjustment: Amount,
+	) -> DispatchResult {
+		Self::adjust_position(who, liquid_id, liquid_adjustment, debit_adjustment)?;
+		loans::Pallet::<T>::update_loan(who, staking_id, -staking_adjustment, -debit_adjustment)
```

Comment:
The parameter type is `Amount`, and our purpose is to convert the collateral KSM/DOT into LKSM/LDOT. 

Should ensure `liquid_adjustment`, `staking_adjustment` and `debit_adjustment` are positive numbers or zero here, because if there are any negative numbers among these, the operation will be reversed and some risk checks may be bypassed, which will be dangerous. It is recommended to use `Balance` as the parameter type although the type conversion will be troublesome

Reply:
Yeah I agree it makes a lot more sense to pass `Balance` in the arguments and do the type conversion there instead of the `checked_into` currently done in the `mint_from_cdp_loan`.

In general I don't love how both `transfer_collateral_from_loan` and `swap_position_to_liquid` are unsafe individually and only work together but I don't see a better way